### PR TITLE
[12.0][FIX]  product_supplierinfo_for_customer: assure you get partner id

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Minor fix. Please, fast-tracking.

### CODE REVIEW

File: product_supplierinfo_for_customer/models/product_product.py [hunk 1]
```diff
@@ -29,6 +29,8 @@ def price_compute(
         if price_type == 'partner':
             partner = self.env.context.get('partner_id', False) or \
                 self.env.context.get('partner', False)
+            if partner and not isinstance(partner, (int,)):
```
Comment (pedrobaeza) [inline product_supplierinfo_for_customer/models/product_product.py]: Instead of looking for not being integer, better to check if is `BaseModel`

Comment (MiquelRForgeFlow) [reply product_supplierinfo_for_customer/models/product_product.py]: done!

File: product_supplierinfo_for_customer/models/product_product.py [hunk 2]
```diff
@@ -29,6 +29,8 @@ def price_compute(
         if price_type == 'partner':
             partner = self.env.context.get('partner_id', False) or \
                 self.env.context.get('partner', False)
+            if partner and isinstance(partner, models.BaseModel):
+                partner = partner.id
```
Comment (pedrobaeza) [inline product_supplierinfo_for_customer/models/product_product.py]: But if this is an ID, why the variable is called `partner`? It should be `partner_id` (I know that the code may not be yours, but we have now the opportunity to fix this).

Comment (MiquelRForgeFlow) [reply product_supplierinfo_for_customer/models/product_product.py]: done

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**